### PR TITLE
Feature/realm listener

### PR DESCRIPTION
**Pull request by ganfra**
Remove some usage of realm listeners to process the events as they come from sync/pagination/local repo

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/TokenChunkEventPersistor.kt [hunk 1]
```diff
@@ -154,6 +157,9 @@ internal class TokenChunkEventPersistor @Inject constructor(private val monarchy
                         handlePagination(realm, roomId, direction, receivedChunk, currentChunk)
                     }
                 }
+        roomEventsProcessors.process(RoomEventsProcessor.Mode.PAGINATING, roomId, receivedChunk.events)
+        roomEventsProcessors.process(RoomEventsProcessor.Mode.PAGINATING, roomId, receivedChunk.stateEvents)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/TokenChunkEventPersistor.kt]: Maybe process state event first?

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt [hunk 2]
```diff
@@ -106,13 +110,30 @@ internal class SyncResponseHandler @Inject constructor(private val monarchy: Mon
...
+        val processMode = if (isInitialSync) {
+            RoomEventsProcessor.Mode.INITIAL_SYNC
+        } else {
+            RoomEventsProcessor.Mode.INCREMENTAL_SYNC
+        }
+        syncResponse.join.forEach {
+            val roomId = it.key
+            val roomSync = it.value
+            roomEventsProcessors.process(processMode, roomId, roomSync.timeline?.events ?: emptyList())
+            roomEventsProcessors.process(processMode, roomId, roomSync.accountData?.events ?: emptyList())
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt]: accountData events are processed?

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt [hunk 3]
```diff
@@ -106,13 +110,30 @@ internal class SyncResponseHandler @Inject constructor(private val monarchy: Mon
...
+            roomEventsProcessors.process(processMode, roomId, roomSync.ephemeral?.events ?: emptyList())
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt]: ephemeral events are processed?

Comment (ganfra) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt]: Not at the moment, not sure it's relevant to keep account and ephemeral

Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt]: COuld we have 3 methods? 
processTimelineEvent
processRoomAccountDataEvent
processEphemeralEvents

**Comment by bmarty** [comment]
Cleanup will be part of a dedicated PR

**Pull request closed without merging**